src/test/kotlin/org/rust/ide/refactoring/move/RsMoveTopLevelItemsTestBase.kt

```diff
@@ -0,0 +1,94 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.refactoring.move
+
+import com.intellij.openapi.application.runWriteAction
+import com.intellij.openapi.editor.Document
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiDirectory
+import com.intellij.psi.PsiDocumentManager
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.intellij.psi.util.parentOfType
+import org.intellij.lang.annotations.Language
+import org.rust.RsTestBase
+import org.rust.TestProject
+import org.rust.lang.core.psi.ext.RsItemElement
+import org.rust.lang.core.psi.ext.RsMod
+import org.rust.lang.core.psi.ext.childrenOfType
+import org.rust.openapiext.document
+import org.rust.openapiext.toPsiDirectory
+import org.rust.openapiext.toPsiFile
+import org.rust.stdext.mapToSet
+
+abstract class RsMoveTopLevelItemsTestBase : RsTestBase() {
+
+    protected fun doTest(@Language("Rust") before: String, @Language("Rust") after: String) =
+        checkByDirectory(before.trimIndent(), after.trimIndent(), ::performMove)
+
+    private fun performMove(testProject: TestProject) {
+        val sourceFile = myFixture.findFileInTempDir(testProject.fileWithCaret).toPsiFile(project)!!
+        val itemsToMove = sourceFile.getElementsAtMarker().mapToSet { it.parentOfType<RsItemElement>()!! }
+
+        val root = myFixture.findFileInTempDir(".").toPsiDirectory(project)!!
+        val targetMod = searchElementInAllFiles(root) { it.getElementAtMarker(TARGET_MARKER) }
+            ?.parentOfType<RsMod>()
+            ?: error("Please add $TARGET_MARKER marker for target mod")
+
+        RsMoveTopLevelItemsProcessor(
+            project,
+            itemsToMove,
+            targetMod,
+            searchForReferences = true
+        ).run()
+    }
+
+    private fun <T> searchElementInAllFiles(root: PsiDirectory, searcher: (PsiFile) -> T?): T? {
```

Comment:
Probably it's better to rewrite this method to something like
```Kotlin
fun <T> searchElementInAllFiles(root: VirtualFile, searcher: (PsiFile) -> T?): T?
```
using `VfsUtilCore.visitChildrenRecursively` (see usages in our other tests to find examples)

What do you think?

Reply:
Thank you, replaced with `visitChildrenRecursively`